### PR TITLE
Feat/business metrics

### PR DESCRIPTION
**Pull request by vgorkavenko**

### CODE REVIEW

File: src/services/validator_state.py [hunk 1]
```diff
@@ -223,9 +232,22 @@ def is_validator_recently_requested_but_not_exited(validator: LidoValidator) ->
...
+                        not validator_recently_requested_to_exit(validator)
+                ):
+                    return True
+
+                return False
+
             validators_recently_requested_to_exit.extend(
                 list(filter(is_validator_recently_requested_but_not_exited, validators))
             )
+            delayed_validators_count = len(list(filter(is_validator_delayed, validators)))
```
Comment (F4ever) [inline src/services/validator_state.py:248]: 5 functions in one method =)
Lets rewrite this part

File: src/services/validator_state.py [hunk 2]
```diff
@@ -223,9 +232,22 @@ def is_validator_recently_requested_but_not_exited(validator: LidoValidator) ->
 
                 return False
 
+            def is_validator_delayed(validator: LidoValidator) -> bool:
+                if (
+                    validator_requested_to_exit(validator) and
+                        not is_on_exit(validator) and
```
Comment (F4ever) [inline src/services/validator_state.py:238]: remove tab

**Review by F4ever — APPROVED** [review]

File: src/web3py/extensions/contracts.py [hunk 3]
```diff
@@ -124,5 +124,13 @@ def get_el_vault_balance(self, blockstamp: BlockStamp) -> Wei:
     @lru_cache(maxsize=1)
     def get_accounting_last_processing_ref_slot(self, blockstamp: BlockStamp) -> SlotNumber:
         result = self.accounting_oracle.functions.getLastProcessingRefSlot().call(block_identifier=blockstamp.block_hash)
-        FRAME_LAST_REPORT_REF_SLOT.set(result)
+        FRAME_PREV_REPORT_REF_SLOT.set(result)
+        return result
+
+    @lru_cache(maxsize=1)
```
Comment (F4ever) [inline src/web3py/extensions/contracts.py:130]: Remove cache here

File: src/web3py/extensions/contracts.py [hunk 4]
```diff
@@ -124,5 +124,13 @@ def get_el_vault_balance(self, blockstamp: BlockStamp) -> Wei:
...
+    def get_ejector_last_processing_ref_slot(self, blockstamp: BlockStamp) -> SlotNumber:
+        result = self.validators_exit_bus_oracle.functions.getLastProcessingRefSlot().call(
+            block_identifier=blockstamp.block_hash
+        )
+        FRAME_PREV_REPORT_REF_SLOT.set(result)
```
Comment (F4ever) [inline src/web3py/extensions/contracts.py:135]: Add log here

File: src/web3py/extensions/contracts.py (hunk 3 above)
Comment (vgorkavenko) [reply src/web3py/extensions/contracts.py]: fixed

File: src/web3py/extensions/contracts.py (hunk 4 above)
Comment (vgorkavenko) [reply src/web3py/extensions/contracts.py:136]: done

**Pull request merged**